### ISSUE TITLE
Using an external theme?

### ISSUE THREAD

**Issue by jasmcaus** [NvChad/NvChad#347]
Not a bug per se, but I was wondering how do I go about using an external theme not included with NvChad. I'm new to Lua and Nvim's configs with Lua, but I'm interested in using the Github theme for Neovim (https://github.com/projekt0n/github-nvim-theme). 

They mention using this in packer.
```lua
use 'projekt0n/github-nvim-theme'
``` 
and 
```lua
require('github-theme').setup()
```

How do I disable the themes included by default in NvChad, and then proceed to using the above mentioned theme?

**Comment by G-Rowell** [comment]
One of the problems with this would be plugin theming,
often plugins have their own colour settings that need to be configured for each theme.

My understanding is that NvChad does this where many `plug and play` themes don't,
a core part of this functionality is: https://github.com/NvChad/nvim-base16.lua

If you really wanted to do this, my suggestion would probably be to make a PR to the above repo with the theme you want.
Otherwise to fork this repo and start reading the src code & modify it appropriately.

@siduck76 Correct me if I'm wrong>

@jasmcaus Let me know if this is good to close? Feel free to ask more questions ofc!!!

**Comment by jasmcaus** [comment]
Okay, will check this out and let you know :)

**Comment by siduck76** [comment]
I agree with @G-Rowell , it would be better to port that theme for nvchad

**Comment by ashincoder** [comment]
I think we need more highlights like the headings on markdown filetypes .. etc.

**Comment by siduck76** [comment]
> I think we need more highlights like the headings on markdown filetypes .. etc.

yes show the highlight groups so ima add it.

**Comment by siduck76** [comment]
> I think we need more highlights like the headings on markdown filetypes .. etc.

or wait for this https://github.com/nvim-treesitter/nvim-treesitter/issues/872

**Comment by ashincoder** [comment]
Ok..
```lua
markdownCode 
markdownCodeBlock
markdownH1 
markdownH2 
markdownLinkText 

```

**Comment by ashincoder** [comment]
```
Tag 
Link
URL 
Underlined
SpecialComment
```

**Comment by siduck76** [comment]
Lul these are a lot , I will put this on hold until I add treesitter highlights oki

**Comment by Shivansh-007** [comment]
Having to add every theme as a nvchad theme is not the best to go about this, you can't add every single theme here. In my opinion, there should be support for using an external theme.

**Comment by siduck76** [comment]
> Having to add every theme as a nvchad theme is not the best to go about this, you can't add every single theme here. In my opinion, there should be support for using an external theme.

there is already lul , but just dont expect the ui like statusline etc to match that theme then :/

**Comment by G-Rowell** [comment]
@Shivansh-007 

Take a look at this file: https://github.com/NvChad/NvChad/blob/main/lua/colors/highlights.lua

This is one of the reasons why NvChad needs to have a theme in a given format.

NvChad does a lot of manual work for you by setting the colours of all the plugins.

Without this any theme would just change the base NeoVim but not plugin colours.

So you can go ahead and do so already, but it provides very poor colour theming.

I do understand your point though, but the file linked above is why such a request is not so easy to do.


G

**Comment by devth** [comment]
I'm kinda confused by this. Can't custom UI like statusline pull colors from a configured colorscheme? Why does it have to be a custom thing?

**Comment by siduck** [comment]
> I'm kinda confused by this. Can't custom UI like statusline pull colors from a configured colorscheme? Why does it have to be a custom thing?

you can. But its all upto you, if youre good with lua and nvim stuff then its gonna be easy to even use external themes. 

First you have to remove the base16 theme 

https://nvchad.github.io/config/Custom%20config#remove--redefine-plugins

then override default hightlights 

https://nvchad.github.io/config/Custom%20config#override-default-highlights

**Comment by ss77a** [comment]
The nvchad docs are don't cover some things in-depth. I've found some documentation from other forms https://docs.rockylinux.org/books/nvchad/install_nvchad/ but it might be outdated slightly

I'm hoping I can use override to nullify a lot of the themes. I've been playing around with config. Looking at adding things.

I'm wondering if there is the possibility to override themes. Is there a syntax to achieve that?

**Comment by siduck** [comment]
@ss77a check nvchad.com ( theming section )

**Comment by pencilcheck** [comment]
so still not possible to have external theme integration?

**Comment by siduck** [comment]
@pencilcheck it is but too much work. at this point why even use nvchad? use other distros. Base46 & UI plugins are the core of nvchad. You can add local themes in base46 format tho

**Comment by pencilcheck** [comment]
> @pencilcheck it is but too much work. at this point why even use nvchad? use other distros. Base46 & UI plugins are the core of nvchad. You can add local themes in base46 format tho

why so defensive? is there is a way to automatically convert external themes to compatible base46 stuff that nvchad can use i'm totally fine by that.

**Comment by siduck** [comment]
> > @pencilcheck it is but too much work. at this point why even use nvchad? use other distros. Base46 & UI plugins are the core of nvchad. You can add local themes in base46 format tho
> 
> why so defensive? is there is a way to automatically convert external themes to compatible base46 stuff that nvchad can use i'm totally fine by that.

because base46 is the main feature of nvchad, why would you use   nvchad if not base46

**Comment by pencilcheck** [comment]
> > > @pencilcheck it is but too much work. at this point why even use nvchad? use other distros. Base46 & UI plugins are the core of nvchad. You can add local themes in base46 format tho
> > 
> > 
> > why so defensive? is there is a way to automatically convert external themes to compatible base46 stuff that nvchad can use i'm totally fine by that.
> 
> because base46 is the main feature of nvchad, why would you use nvchad if not base46

if there is no way then there is no way, I would imagine this is not that hard to have a simple way to convert/import those into base46.

**Comment by siduck** [comment]
> > > > @pencilcheck it is but too much work. at this point why even use nvchad? use other distros. Base46 & UI plugins are the core of nvchad. You can add local themes in base46 format tho
> > > 
> > > 
> > > why so defensive? is there is a way to automatically convert external themes to compatible base46 stuff that nvchad can use i'm totally fine by that.
> > 
> > 
> > because base46 is the main feature of nvchad, why would you use nvchad if not base46
> 
> if there is no way then there is no way, I would imagine this is not that hard to have a simple way to convert/import those into base46.

automated colorschemes dont look that good. you can check https://github.com/NvChad/pywal 

and FYI not all themes store their palette colors in same filestructure , you'll have to install that theme and run a function that extracts all hues
